test/t8_schemes/t8_gtest_successor.cxx

```diff
@@ -35,24 +54,25 @@ t8_recursive_successor (t8_element_t *element, t8_element_t *successor,
                         const int maxlvl)
 {
   int                 level = ts->t8_element_level (element);
-  T8_ASSERT (ts->t8_element_level (element) <= maxlvl
-             && maxlvl <= ts->t8_element_maxlevel () - 1);
-  int                 num_children, i;
-  num_children = ts->t8_element_num_children (element);
+  ASSERT_TRUE (ts->t8_element_level (element) <= maxlvl
+               && maxlvl <= ts->t8_element_maxlevel () - 1);
+  int                 num_children = ts->t8_element_num_children (element);
```

Comment:
```suggestion
  const int                 num_children = ts->t8_element_num_children (element);
```
I think num_children does not change after this line and it can be const. Please double-check. If so, we can delete line 68 and line 88, see my following suggestions.